rust/Cargo.toml

```diff
@@ -35,9 +35,9 @@ categories = ["database"]
 
 [workspace.dependencies]
 adbc_core = { path = "./core" }
-arrow-array = { version = "53.1.0", default-features = false, features = [
+arrow-array = { version = "53.4.0", default-features = false, features = [
     "ffi",
 ] }
 arrow-buffer = { version = "53.1.0", default-features = false }
 arrow-schema = { version = "53.1.0", default-features = false }
```

Comment:
@lidavidm @alamb do you know why these didn't get upgraded to `53.4.0` as well?

Reply:
Those crates were upgraded (see `Cargo.lock`). Looks like dependabot handles this a bit different, most likely because these crates are direct dependencies of `arrow-array`.

Reply:
I don't know why they didn't get updated in Cargo.toml

The current contents basically mean 53.1.0 or later (53.2.0 or 53.3.0)

Reply:
Taking a wild guess: for these groups dependabot checks if the lock file is outdated, and after bumping `arrow-array` it is up to date for those other dependencies (they are dependencies of `arrow-array`, so they get bumped too in the lock file as a result of bumping `arrow-array`), so it skips bumping those in the `Cargo.toml`.

In the end it doesn't really matter here; keeping the minor version low here just means we don't rely on features introduced in higher compatible versions. By updating the lock file we are just making sure things are building fine in CI with the latest version.

Reply:
> By updating the lock file we are just making sure things are building fine in CI with the latest version.

Yeah, the `Cargo.lock` file has locked on `53.4.0` for every dependency. Even the ones that say `_.1._` in the `Cargo.toml`.